### PR TITLE
[SOAR-13808] Insight Connect SDK - New helper functions added

### PR DESCRIPTION
**Pull request by igorski-r7**
### Description
Moved some reusable functions from plugins to the SDK `helper.py` file.

### Motivation and Context
Not to duplicate so many code between many plugins (same as copying the same function bodies in order to use them in different plugin).

### How Has This Been Tested?
Copied-pasted those function from currently working plugins.

### Types of changes

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to change)

### Checklist:

- [X] I have updated the documentation accordingly (if changes are required).
- [X] I have tested the changes locally.

### CODE REVIEW

File: insightconnect_plugin_runtime/helper.py [hunk 1]
```diff
@@ -113,8 +199,121 @@ def get_hashes_string(s):
...
+        output_dict = {}
+        for key, value in to_modify.items():
+            output_dict[case_method(key)] = convert_keys_to_camel(value)
+        return output_dict
+    else:
+        return to_modify
+
+
+def backoff_function(attempt: int) -> float:
+    """backoff_function. Back-off function used in rate_limiting retry decorator
```
Comment (cmcnally-r7) [inline insightconnect_plugin_runtime/helper.py:233]:
```suggestion
    """
    Back-off function used in rate_limiting retry decorator
```

File: insightconnect_plugin_runtime/helper.py [hunk 2]
```diff
@@ -113,8 +199,121 @@ def get_hashes_string(s):
...
+    :rtype: float
+    """
+
+    return 2 ** (attempt * 0.6)
+
+
+def rate_limiting(
+    max_tries: int, back_off_function: Callable = backoff_function
+) -> Union[dict, None]:
+    """rate_limiting. This decorator allows to work API call with rate limiting by using exponential backoff function.
```
Comment (cmcnally-r7) [inline insightconnect_plugin_runtime/helper.py:248]:
```suggestion
    """
    This decorator allows to work API call with rate limiting by using exponential backoff function.
```

**Review by cmcnally-r7 — COMMENTED** [review]
Looks good to me just a couple of new lines and such to tidy up the comments a bit if you want 😄

**Review by hpwilson-r7 — APPROVED** [review]
Looks awesome to me!

**Review by TKuzas-R7 — DISMISSED** [review]

**Review by cmcnally-r7 — DISMISSED** [review]

File: insightconnect_plugin_runtime/helper.py [hunk 3]
```diff
@@ -4,15 +4,26 @@
...
 from io import IOBase
+from typing import Any, Callable, Dict, List, Union
 from urllib import request
-import shlex
 
 import requests
 
+from insightconnect_plugin_runtime.exceptions import PluginException
+
+SNAKE_CASE_REGEX = r"\b[a-z0-9]+(_[a-z0-9]+)*\b"
```
Comment (TKuzas-R7) [inline insightconnect_plugin_runtime/helper.py:20]: it looks like this is not being used

**Review by TKuzas-R7 — APPROVED** [review]

**Review by cmcnally-r7 — APPROVED** [review]

**Review by TKuzas-R7 — DISMISSED** [review]

**Review by dlaverty-r7 — DISMISSED** [review]

**Review by dlaverty-r7 — APPROVED** [review]

**Review by llaszuk-r7 — APPROVED** [review]

**Review by ablakley-r7 — APPROVED** [review]

**Review by cmcnally-r7 — APPROVED** [review]

**Pull request merged**